src/train.py

```diff
@@ -0,0 +1,145 @@
+import argparse
+
+import segmentation_models_pytorch as smp
+import torch
+from lightning import Trainer
+from pip._internal.operations import freeze
+from torch.utils.data import DataLoader
+from torchvision.models import (
+    convnext_base,
+    efficientnet_v2_m,
+    mobilenet_v3_large,
+    resnet50,
+    resnext50_32x4d,
+    swin_b,
+    vgg16,
+    vit_b_16,
+)
+
+from src import log
+from src.callbacks import BenchmarkCallback
+from src.data.in_memory_dataset import InMemoryDataset
+from src.models.lightning_modules import LitClassification
+
+logger = log.setup_custom_logger()
+
+
+ARCHITECTURES = {
+    "resnet50": resnet50,
+    "convnext": convnext_base,
+    "vgg16": vgg16,
+    "efficient_net_v2": efficientnet_v2_m,
+    "mobilenet_v3": mobilenet_v3_large,
+    "resnext50": resnext50_32x4d,
+    "swin": swin_b,
+    "vit": vit_b_16,
+    "unet_resnet50": smp.Unet
+    # TODO"ssd_vgg16": ssd300_vgg16,
+    # TODO "fasterrcnn_resnet50_v2": fasterrcnn_resnet50_fpn_v2,
+}
+
+
+def print_requirements():
+    pkgs = freeze.freeze()
+    for pkg in pkgs:
+        logger.info(pkg)
+
+
+def main(args):
+    if args.list_requirements:
+        print_requirements()
+
+    args_dict = vars(args)
+    logger.info(f"User Arguments {args_dict}")
+
+    dataset = InMemoryDataset(width=args.width, height=args.width)
+    data_loader = DataLoader(
+        dataset,
+        num_workers=args.n_workers,
+        batch_size=args.batch_size,
+        shuffle=False,
+        pin_memory=True,
+        drop_last=True,
+    )
+
+    trainer = Trainer(
+        accelerator=args.accelerator,
+        strategy="ddp",
+        precision=args.precision,
+        max_steps=args.n_iters + args.warmup_steps,
```

Comment:
i think max_steps is a very ugly way to determine when to stop because then the tqdm is never meant to get filled and you have no idea how long ur exp will last